src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -3833,6 +3832,17 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
             } else if (t->type == ASR::ttypeType::Character) {
                 fmt.push_back("%s");
                 args.push_back(tmp);
+            } else if (ASRUtils::is_logical(*t)) {
+                int ivalue = down_cast<ASR::LogicalConstant_t>(
+                                       ASRUtils::expr_value(v))->m_value;
+                std::string logical_str;
+                if (ivalue == 1) {
+                    logical_str = "True";
+                } else {
+                    logical_str = "False";
+                }
+                llvm::Value *fmt_ptr = builder->CreateGlobalStringPtr(logical_str);
```

Comment:
I think this will work only if `v` is a constant? What if `v` is a variable? I think we should generate the `if-else` check in LLVM (should be easy to do) to select a LLVM string to be stored in `fmt_ptr`.

Reply:
Yes, this will only work at compile time. One has to pass in the strings `True`/`False` for Python and `.true.`/`.false.` for Fortran at compile time, and then at runtime select true or false, using either a C function or some logic in LLVM.

Reply:
@namannimmo10 Are you still working on this?

Reply:
Thanks for the ping and for reviewing the PR.  Please feel free to move with this; I'm focusing more on refactoring the ASR as of now. If you don't have time, I can come back to it this week.